src/recipe/file.service.ts

```diff
@@ -0,0 +1,32 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';
+
+@Injectable()
+export class FileService {
```

Comment:
I would go with a separate service for S3 service that would establish configuration (secrets, keys etc) and expose a method to upload the file (directly using `S3Client`)

The other service would be FileService that would be responsible for business logic of file processing. And would call the previous service to upload the files.

**Important note. Service that handles s3 configuration should be mocked!!!**

Reply:
I think it is done